Lighting/Src/lighting_controller.cpp

```diff
@@ -37,122 +38,69 @@ uint8_t bank_output_buffer[BANK_OUTPUT_BUFFER_SIZE];
 
 WS2812 leds[NUM_LEDS]; // TODO: make this work
 
-// Initial setup call
-LightingController rev3(dma_output_buffer, bank_output_buffer, leds); // TODO: Once we have custom functions registered as callbacks.....
+//list of colours
+RGB_colour_t WHITE = { 255, 255, 255 };
+RGB_colour_t RED = { 255, 0, 0 };
+RGB_colour_t ORANGE = {255, 165, 0};
+RGB_colour_t GREEN = {0, 255, 0};
+RGB_colour_t CYAN = {0, 255, 255};
+RGB_colour_t BROWN = {139, 69, 19};
+RGB_colour_t PURPLE = {255, 0, 255};
 
-extern TIM_HandleTypeDef htim;
+// Initial setup call
+LightingController rev4(dma_output_buffer, bank_output_buffer, leds, &htim1, TIM_CHANNEL_2); // TODO: Once we have custom functions registered as callbacks.....
 
+// Temporary (ish) function with exemplar code that allows us to test lighting board functionality without needing CAN commands
 void run_lighting_board() {
 
 	// Call to start lighting control
-	rev3.start_lighting_control();
-
-	// RGB Colour & Brightness control demo setup
-
-	RGB_colour_t my_colour;
-	my_colour.red = 65;
-	my_colour.green = 0;
-	my_colour.blue = 127;
-
-	int red_direction = 1;   // 1 for increasing, -1 for decreasing
-	int green_direction = 1; // 1 for increasing, -1 for decreasing
-	int blue_direction = 1;  // 1 for increasing, -1 for decreasing
-
-	uint8_t brightness_slow = 0;
-	int brightness_direction = 1;
-	uint8_t brightness = 0;
+	rev4.start_lighting_control();
+	HAL_TIM_Base_Start_IT(&htim2);
 
 	// DOMAIN SETUP
 	// TODO: move Control Domain building to special functions
 
-	RGB_colour_t WHITE = { 255, 255, 255 };
-	RGB_colour_t RED = { 255, 0, 0 };
-	uint8_t BRIGHTNESS_MAX = 100;
-
-	// build beacon domain
-	rev3.set_domain_colour(CD_BEACON, RED, BRIGHTNESS_MAX);
-	rev3.add_led_to_cd(1, CD_BEACON);
-	rev3.add_led_to_cd(4, CD_BEACON);
-
-	// build strobe domain
-	rev3.set_domain_colour(CD_STROBE, WHITE, BRIGHTNESS_MAX);
-	rev3.add_led_to_cd(0, CD_STROBE);
-	rev3.add_led_to_cd(2, CD_STROBE);
-	rev3.add_led_to_cd(3, CD_STROBE);
-	rev3.add_led_to_cd(5, CD_STROBE);
-
+	// allow all of our domains
+	// comment any of these out to see the effect of allowing command domains
+	rev4.allow_domain(CD_MAIN);
+	rev4.allow_domain(CD_TAXI);
+	rev4.allow_domain(CD_BEACON);
+	rev4.allow_domain(CD_STROBE);
+	rev4.allow_domain(CD_LANDING);
+	rev4.allow_domain(CD_NAVIGATION);
+	rev4.allow_domain(CD_BRAKE);
+	rev4.allow_domain(CD_SEARCH);
```

Comment:
Looking at this, it might be nice to have a function `allow_domains(uint8_t allowed_domains)` that simply accepts a bitfield of allowed domains and sets the domains accordingly.

This is very explicit and very good, but just for our initial setup calls, it might make life a bit easier? Just a thought.